include/sundials/impl/sundials_mpi_errors_impl.h

```diff
@@ -0,0 +1,85 @@
+/* -----------------------------------------------------------------
+ * SUNDIALS Copyright Start
+ * Copyright (c) 2002-2022, Lawrence Livermore National Security
+ * and Southern Methodist University.
+ * All rights reserved.
+ *
+ * See the top-level LICENSE and NOTICE files for details.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ * SUNDIALS Copyright End
+ * -----------------------------------------------------------------
+ * !!!!!!!!!!!!!!!!!!!!!!!!! WARNING !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+ * This is a 'private' header file and should not be used in user
+ * code. It is subject to change without warning.
+ * !!!!!!!!!!!!!!!!!!!!!!!!! WARNING !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+ * -----------------------------------------------------------------
+ * Contains all error checking macros and private error handling API.
+ * -----------------------------------------------------------------*/
+
+#ifndef _SUNDIALS_MPI_ERRORS_IMPL_H
+#define _SUNDIALS_MPI_ERRORS_IMPL_H
+
+#include <sundials/sundials_mpi_errors.h>
+
+#if !defined(SUNDIALS_DISABLE_ERROR_CHECKS)
+#define SUNCheckMPICall(call)                                               \
+  do {                                                                      \
+    int sun_chk_mpi_call_err_code_ = call;                                  \
+    if (sun_chk_mpi_call_err_code_ != MPI_SUCCESS)                          \
```

Comment:
I keep looking to see how `mpi.h` gets included into this file (so that `MPI_SUCCESS` exists), but I can't seem to find it.  Would it be preferable to include it here directly, instead of depending on someone else to include it 2-3 layers above?

Reply:
Yeah it should be included here, I just forgot.